config/vector.spec.toml

```diff
@@ -4470,6 +4470,174 @@ end
   healthcheck = true
   healthcheck = false
 
+# Batches `log` events to Sematext via the Elasticsearch API.
+[sinks.sematext]
+  #
+  # General
+  #
+
+  # The component type. This is a required field that tells Vector which
+  # component to use. The value _must_ be `sematext`.
+  #
+  # * required
+  # * type: string
+  # * must be: "sematext"
+  type = "sematext"
+
+  # A list of upstream source or transform IDs. See configuration for more info.
+  #
+  # * required
+  # * type: [string]
+  inputs = ["my-source-id"]
+
+  # The cloud destination to send logs too.
+  #
+  # * required
+  # * type: string
+  cloud = "north_america"
+  cloud = "europe"
+
+  # The token that will be used to write to Sematext.
+  #
+  # * required
+  # * type: string
+  token = "some-sematext-token"
+
+  # Enables/disables the sink healthcheck upon start.
+  #
+  # * optional
+  # * default: true
+  # * type: bool
+  healthcheck = true
+  healthcheck = false
+
+  #
+  # Batch
+  #
+
+  [sinks.sematext.batch]
+    # The maximum size of a batch before it is flushed.
+    #
+    # * optional
+    # * default: 10490000
+    # * type: int
+    # * unit: bytes
+    max_size = 10490000
+
+    # The maximum age of a batch before it is flushed.
+    #
+    # * optional
+    # * default: 1
+    # * type: int
+    # * unit: seconds
+    timeout_secs = 1
+
+  #
+  # Buffer
+  #
+
+  [sinks.sematext.buffer]
+    # The buffer's type / location. `disk` buffers are persistent and will be
+    # retained between restarts.
+    #
+    # * optional
+    # * default: "memory"
+    # * type: string
+    # * enum: "memory" or "disk"
+    type = "memory"
+    type = "disk"
+
+    # The maximum number of events allowed in the buffer.
+    #
+    # * optional
+    # * default: 500
+    # * type: int
+    # * unit: events
+    # * relevant when type = "memory"
+    max_events = 500
```

Comment:
For Sematext you could make this bigger (more efficient), like 5000.  The batch max_size will cap it to no more than 10 MB by default, IIUC.

Reply:
This is just for in memory buffer and isn't related to actual batch size (which is basically another in memory buffer). It may make sense to raise the max batch size instead?

Reply:
Aha, I see.  10 MB is fine for batch size.  I thought you had the model where you specify 2 limits - size in # of events and size in bytes, and then flush when either limit is reached (or maybe you also have something like "max time between flushes").  But IIUC you only flush based batch size in bytes.

Reply:
Right so we have "batches" which we use for how we bundle events together to send them in requests. In most cases we do this via bytes but some API's prefer to impose their limits via the amount of actual events. Cloudwatch is a good example, in those cases its based on the max events.

The "buffer" stuff is related to when the downstream service is applying back pressure and we need to store events while the downstream isn't accepting batches. In this case if you choose the memory buffer we buffer based on the amount of events. If you pick disk it will buffer based on the on disk size that we are using.